src/lib/async/wait.ts

```diff
@@ -0,0 +1,5 @@
+export async function wait<T>(delay: number, fn: T): Promise<Awaited<T>> {
+  return await Promise.all([fn, new Promise(y => setTimeout(y, delay))]).then(
+    arr => arr[0],
+  )
+}
```

Comment:
This is a util to provide a _minimum_ execution time. Helpful for load states and things.